base/workflow/inst/batch_runs.R

```diff
@@ -0,0 +1,120 @@
+  settings <- list()
+  
+  # Info
+  settings$info$notes <- paste0("Test_Run")
+  settings$info$userid <- user_id
+  settings$info$username <- "None"
+  settings$info$dates <- Sys.Date()
+  #Outdir
+  model.new <- tbl(bety, "models") %>% filter(model_id == id) %>% collect()
+  outdir_base<-config.list$output_folder
+  outdir_pre <- paste(model.new$model_name,format(as.Date(start_date), "%Y-%m"),
+                      format(as.Date(end_date), "%Y-%m"),
+                      met,site_id,"_test_runs",
+                      sep="",collapse =NULL)
+  outdir <-  paste0(outdir_base,outdir_pre)
+  dir.create(outdir)
+  settings$outdir <- outdir
+  #Database BETY
+  settings$database$bety$user <- config.list$db_bety_username
+  settings$database$bety$password <- config.list$db_bety_password
+  settings$database$bety$host <- config.list$db_bety_hostname
+  settings$database$bety$dbname <- config.list$db_bety_database
+  settings$database$bety$driver <- "PostgreSQL"
+  settings$database$bety$write <- FALSE
+  #Database dbfiles
+  settings$database$dbfiles <- config.list$dbfiles_folder
+  #PFT
+  if (is.na(pft_name)){
+    pft <- tbl(bety, "pfts") %>% filter(modeltype_id == model.new$modeltype_id) %>% collect()
+    pft_name <- pft$name[1]
+  }
+  settings$pfts$pft$name <- pft_name
+  settings$pfts$pft$constants$num <- 1
+  #Meta Analysis
+  settings$meta.analysis$iter <- 3000
+  settings$meta.analysis$random.effects <- FALSE
+  #Ensemble
+  settings$ensemble$size <- 1
+  settings$ensemble$variable <- "GPP"
+  settings$ensemble$samplingspace$met$method <- "sampling"
+  settings$ensemble$samplingspace$parameters$method <- "uniform"
+  #Model
+  settings$model$id <- model.new$id
+  #Worflow
+  settings$workflow$id <- paste0("Test_run_","_",model.new$model_name)
+  settings$run$site$id <- site_id
+  settings$run$site$met.start <- start_date
+  settings$run$site$met.end <- end_date
+  settings$run$inputs$met$source <- met
+  settings$run$inputs$met$output <- model.new$model_name
+  settings$run$inputs$met$username <- "pecan" 
+  settings$run$start.date <- start_date
+  settings$run$end.date <- end_date
+  settings$host$name <-config.list$db_bety_hostname
+  
+  #create file and Run
+  saveXML(listToXml(settings, "pecan"), file=paste0(outdir,"/","pecan.xml"))
+  file.copy(paste0(config.list$pecan_home,"web/","workflow.R"),to = outdir)
+  setwd(outdir)
+  ##Name log file
+  log <- file("workflow.Rout", open = "wt")
+  sink(log)
+  sink(log, type = "message")
+  source("workflow.R")
+  sink()
+}
+
+
+
+
+##Create Run Args
+pecan_path <- "/fs/data3/tonygard/work/pecan"
+config.list <- PEcAn.utils::read_web_config(paste0(pecan_path,"/web/config.php"))
+bety <- betyConnect(paste0(pecan_path,"/web/config.php"))
+con <- bety$con
+
+## Find name of Machine R is running on
+mach_name <- Sys.info()[[4]]
+mach_id <- tbl(bety, "machines")%>% filter(grepl(mach_name,hostname)) %>% pull(id)
+  
+model_ids <- tbl(bety, "dbfiles") %>% filter(machine_id == mach_id) %>% 
```

Comment:
@tonygardella should we assume all models will work or that we may have models "registered" with a machine but that it may not work at the moment and instead we provide a list of specific models to test OR automatically generate based on BETY

Reply:
It's not the assumption that all models work. Part of the reason for making this is to easily identify what's not working. that being said I could make this part into an if statement where you could provide a vector of model ids and if you don't then it will run all models on your machine.

I'm beginning to see this this could be used for not just testing, but an easy way to do a batch of experiments. I'll take that into account as I continue defining args and adding options.